### PR TITLE
Make build-image Dockerfile multi-platform

### LINKED ISSUE #6084: build-image is amd64 only
This means much slower execution on other platforms that have to emulate amd64
https://github.com/vmware-tanzu/velero/blob/73f17404078f0247c5d27af3890d8f33b598ff6c/hack/build-image/Dockerfile#L15

### PR DESCRIPTION
**Pull request by kaovilai**
Fix #6084

Signed-off-by: Tiger Kaovilai <tkaovila@redhat.com>

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Comment by kaovilai** [comment]
Unstale

**Review by ywk253100 — DISMISSED** [review]

File: hack/build-image/Dockerfile [hunk 1]
```diff
@@ -39,26 +39,59 @@ RUN go install golang.org/x/tools/cmd/goimports@11e9d9cc0042e6bd10337d4d2c3e5d92
...
+#   {{- title .Os }}_
+#   {{- if eq .Arch "amd64" }}x86_64
+#   {{- else if eq .Arch "386" }}i386
+#   {{- else }}{{ .Arch }}{{ end }}
+#   {{- if .Arm }}v{{ .Arm }}{{ end -}}
+RUN ARCH=$(go env GOARCH) && \
+    if [ "$ARCH" = "amd64" ] ; then \
+        ARCH="x86_64"; \
+    elif [ "$ARCH" = "386" ] ; then \
+        ARCH="i386"; \
```
Comment (kaovilai) [inline hack/build-image/Dockerfile:84]: I was gonna add this to support arm32v7 but alas [kubebuilder](https://[storage.googleapis.com/kubebuilder-tools/?prefix=kubebuilder-tools-1.22.1](https://storage.googleapis.com/kubebuilder-tools/?prefix=kubebuilder-tools-1.22.1)) does not have anything below v8.. so stopped here :)
```suggestion
        ARCH="i386"; \
    elif [ "$ARCH" = "arm" ] ; then \
        ARCH=$ARCHv$(go env GOARM); \
```

**Comment by kaovilai** [comment]
validated on linux/
- amd64
- arm64
- ppc64le

known breakage linux/s390x due to missing arch from kubebuilder which [was just resolved last week](https://github.com/kubernetes-sigs/kubebuilder/issues/2298) if and when [we decide to bump version to v3.14.0](https://github.com/kubernetes-sigs/kubebuilder/releases/tag/v3.14.0#:~:text=kubebuilder_linux_s390x)

**Review by ywk253100 — APPROVED** [review]

**Review by sseago — APPROVED** [review]

**Review by blackpiglet — APPROVED** [review]

**Pull request merged**